bigtop-tests/smoke-tests/hive/HiveJdbcGeneralTest.java

```diff
@@ -0,0 +1,392 @@
+      updateTable(stmt, "Insert into table btestv SELECT Dest from btest");
+      updateTable(stmt,
+          "Insert into table btestp PARTITION (Capacity) SELECT * from btest");
+      deleteFile(stmt, filePath + ".txt", HdfsURI);
+      deleteFile(stmt, filePath + "0.orc", HdfsURI);
+      assertEquals("302", printResults(stmt, "Select * from btest"));
+      assertEquals("114", printResults(stmt,
+          "Select * from btest where Dest = 'LAX' order by boarded desc"));
+      assertEquals("622", printResults(stmt,
+          "Select * from btest where boarded between 160 and 180 order by boarded asc"));
+      assertEquals("202", printResults(stmt,
+          "Select * from btest where Dest='LAX' or Dest='ORD'"));
+      assertEquals("114", printResults(stmt,
+          "Select * from btest where Dest= 'LAX' and boarded >= 180"));
+      assertEquals("114", printResults(stmt,
+          "Select * from btest where Dest ='LAX' and boarded = 197"));
+      assertEquals("219",
+          printResults(stmt, "Select * from btest LIMIT 10 --"));
+      assertEquals("114", preparedStatement(con,
+          "Select * from btest where Dest ='LAX' and boarded = 197"));
+      assertEquals(15, setFetchSizeStatement(stmt));
+      assertEquals(15, setFetchSizePreparedStatement(con));
+      // assertEquals(callableStatement(con, -20), -1);
+      assertEquals("95487", printResults(stmt, "Select SUM(Miles) from btest"));
+      assertEquals("302", printResults(stmt, "Select * from btest"));
+      assertEquals(0, updateTable(stmt,
+          "Update btestt set Orig= 'test' where Dest= 'LAX'"));
+      try {
+        loadData(stmt, filePath + ".txt", newTableName + "T");
+        fail("shouldn't get here");
+      } catch (HiveSQLException e) {
+        System.out.println("File does not exist in specified location");
+      }
+      assertEquals("132",
+          printResults(stmt, "Select * from btestt order by Dest"));
+      assertEquals("capacity=250.0",
+          printResults(stmt, "show partitions btestp"));
+      assertEquals("LAX", printResults(stmt,
+          "select MIN(Dest), boarded from btest where Dest ='LAX' group by Dest, boarded"));
+      assertEquals("19", printResults(stmt,
+          "select count(*) from btest where capacity = 250.0 group by capacity"));
+      executeStatement(stmt, "Drop view testview");
+      executeStatement(stmt,
+          "Create view testview AS select SUM(btest.boarded) BOARDED from btest, btestv where btest.Dest=btestv.Dest");
+      assertEquals("45072", printResults(stmt, "Select * from testview"));
+      executeStatement(stmt, "Drop view testview");
+      printResults(stmt, "Describe formatted btest");
+      dropTable(stmt, newTableName);
+      dropTable(stmt, newTableName + "NT");
+      dropTable(stmt, newTableName + "T");
+      dropTable(stmt, newTableName + "P");
+      dropTable(stmt, newTableName + "V");
+      setNegativeFetchSize(stmt);
+    }
+  }

```

Comment:
It seems that we're testing many cases here. Can we separate them one by one? That way we can precisely get which part is failing. If the whole series of operation is what you'd like to test. Then we should rename the test name or add comments to describe what we're going to test here.